src/main/scala/com/fulcrumgenomics/vcf/DownsampleVcf.scala

```diff
@@ -0,0 +1,281 @@
+      else if (bb < ab && bb < aa) IndexedSeq(alleleB, alleleB)
+      else IndexedSeq(alleleA, alleleB)
+    }
+    gt.copy(attrs = Map("PL" -> IndexedSeq(aa, ab, bb), "AD" -> newAds, "DP" -> newAds.sum), calls = calls)
+  }
+
+  /**
+   * Compute the genotype likelihoods given the allele depths.
+   * @param ads The allele depths to generate likelihoods from
+   * @return a list of three likelihoods
+   */
+  def computePls(ads: IndexedSeq[Int]): IndexedSeq[Int] = {
+    val likelihoods = Likelihoods(ads(0), ads(1))
+    IndexedSeq(likelihoods.aa.round.toInt, likelihoods.ab.round.toInt, likelihoods.bb.round.toInt)
+  }
+
+
+  object Likelihoods {
+    /** Computes the likelihoods for each possible genotype.
+     *
+     * @param alleleDepthA the reference allele depth
+     * @param alleleDepthB the alternate allele depth
+     * @param epsilon      the error rate for genotyping
+     * @return a new `Likelihood` that has the likelihoods of AA, AB, and BB
+     */
+    def apply(alleleDepthA: Int, alleleDepthB: Int, epsilon: Double = 0.01): Likelihoods = {
+      val aGivenAA = log10(1 - epsilon)
+      val aGivenBB = log10(epsilon)
+      val aGivenAB = log10((1 - epsilon) / 2)
+
+      val rawGlAA = ((alleleDepthA * aGivenAA) + (alleleDepthB * aGivenBB)) * -10
+      val rawGlBB = ((alleleDepthA * aGivenBB) + (alleleDepthB * aGivenAA)) * -10
+      val rawGlAB = ((alleleDepthA + alleleDepthB) * aGivenAB) * -10
+
+      val minGL = math.min(math.min(rawGlAA, rawGlAB), rawGlBB)
+
+      Likelihoods(
+        aa = rawGlAA - minGL,
+        ab = rawGlAB - minGL,
+        bb = rawGlBB - minGL
+      )
+    }
+  }
+
+  /** Stores the log10(likelihoods) for all possible bi-allelic genotypes.
+   *
+   * @param aa likelihood of AA
+   * @param ab likelihood of AB
+   * @param bb likelihood of BB
+   */
+  case class Likelihoods(aa: Double, ab: Double, bb: Double) {
+    def pls = IndexedSeq(aa.round.toInt, ab.round.toInt, bb.round.toInt)
+  }
+}
+
+  @clp(group = ClpGroups.VcfOrBcf, description =
+    """
+      |DownsampleVcf takes a vcf file and metadata with sequencing info and
+      |1. winnows the vcf to remove variants within a specified distance to each other,
+      |2. downsamples the variants using the provided allele depths and target base count by
+      |   re-computing/downsampling the allele depths for the new target base count
+      |   and re-computing the genotypes based on the new allele depths
+      |and writes a new downsampled vcf file.
+      |For single-sample VCFs, the metadata file can be omitted, and instead you can specify originalBases.
+  """)
+  class DownsampleVcf
+  (@arg(flag = 'i', doc = "The vcf to downsample.") val input: PathToVcf,
+   @arg(flag = 'm', doc = "Index file with bases per sample.") val metadata: Option[FilePath] = None,
+   @arg(flag = 'b', doc = "Original number of bases (for single-sample VCF)") val originalBases: Option[Double] = None,
```

Comment:
I'd add a `--proportion` argument here instead, then make `downsampleToBases` optional for single-sample VCFs.  Thoughts?